eyetracking/bids_scripts/utils.py

```diff
@@ -0,0 +1,372 @@
+
+            if cfd > conf_thresh:
+                clean_x.append(x_norm - x_diff)
+                clean_y.append(y_norm - y_diff)
+                clean_conf.append(cfd)
+                clean_times.append(timestp)
+
+    return reset_gaze_list, (all_x, all_y, all_times, all_conf), (clean_x, clean_y, clean_times, clean_conf)
+
+
+def apply_poly(
+    ref_times: list,
+    distances: list,
+    degree: int,
+    all_times: np.array,
+    anchors: list = [150, 150],
+) -> np.array:
+    '''
+    Fit polynomial to a distribution, then export points along that polynomial curve
+    that correspond to specific gaze time stamps
+
+    The very begining and end of distribution are excluded for stability
+    e.g., participants sometimes look off screen at movie onset & offset,
+    while deepgaze is biased to the center when shown a black screen.
+    This really throws off polynomials
+    '''
+    if degree == 1:
+        p1, p0 = np.polyfit(ref_times[anchors[0]:-anchors[1]], distances[anchors[0]:-anchors[1]], 1)
+        p_of_all = p1*(all_times) + p0
+
+    elif degree == 2:
+        p2, p1, p0 = np.polyfit(ref_times[anchors[0]:-anchors[1]], distances[anchors[0]:-anchors[1]], 2)
+        p_of_all = p2*(all_times**2) + p1*(all_times) + p0
+
+    elif degree == 3:
+        p3, p2, p1, p0 = np.polyfit(ref_times[anchors[0]:-anchors[1]], distances[anchors[0]:-anchors[1]], 3)
+        p_of_all = p3*(all_times**3) + p2*(all_times**2) + p1*(all_times) + p0
+
+    elif degree == 4:
+        p4, p3, p2, p1, p0 = np.polyfit(ref_times[anchors[0]:-anchors[1]], distances[anchors[0]:-anchors[1]], 4)
+        p_of_all = p4*(all_times**4) + p3*(all_times**3) + p2*(all_times**2) + p1*(all_times) + p0
+
+    return p_of_all
+
+
+def apply_gaussian(
+    ref_times: list,
+    distances: list,
+    all_times: np.array,
+    rollwin_dur: float = 15.0,
+    fps: float = 250.0,
+    anchors: list = [150, 150],
+) -> np.array:
+
+    roll_win = int(fps*rollwin_dur)
+
+    dist = distances[anchors[0]:-anchors[1]]
+    median_line = pd.Series(dist).rolling(roll_win).median()
+    under_line = pd.Series(dist).rolling(roll_win).quantile(0.25)
+    over_line = pd.Series(dist).rolling(roll_win).quantile(0.75)
+
+    median_line[:roll_win-1] = np.median(dist[:roll_win])
+    under_line[:roll_win-1] = np.quantile(dist[:roll_win], 0.25)
+    over_line[:roll_win-1] = np.quantile(dist[:roll_win], 0.75)
+
+    # https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_filter.html
+    sig_val = int(fps*20)
+    median_gauss = scipy.ndimage.gaussian_filter(copy.deepcopy(median_line), sigma=sig_val, mode='nearest', truncate=1.0)
+    under_gauss = scipy.ndimage.gaussian_filter(copy.deepcopy(under_line), sigma=sig_val, mode='nearest', truncate=1.0)
+    over_gauss = scipy.ndimage.gaussian_filter(copy.deepcopy(over_line), sigma=sig_val, mode='nearest', truncate=1.0)
+    avg_smooth = (median_gauss + under_gauss + over_gauss)/3
```

Comment:
eerrr, what is that statistical function?? average of quantiles what!?
The function name is quite misleading!